### PR TITLE
[TMVA] Throw an exception when input file for using RBDT does not exist.

### LINKED ISSUE #9316: [TMVA] Experimental::RBDT segfaults if input file does not exist
Running

```cpp
// repro.C
void repro() {
  TMVA::Experimental::RBDT<> bdt("Bc2TauNu_BDT", "doesnotexist.root");
}
```
as

```
root -l -b -q repro.C
```
yields:
```
Error in <TFile::TFile>: file /tmp/doesnotexist.root does not exist

 *** Break *** segmentation violation
```

RBDT should probably protect nullptr accesses or similar and throw an exception or return an error rather than crashing.

### PR DESCRIPTION
**Pull request by lmoneta**
This PR fixes #9316

### CODE REVIEW

File: tmva/tmva/inc/TMVA/RBDT.hxx [hunk 1]
```diff
@@ -48,6 +48,9 @@ public:
    {
       // Get number of output nodes of the forest
       auto file = TFile::Open(filename.c_str(), "READ");
+      if (!file) {
```
Comment (eguiraud) [inline tmva/tmva/inc/TMVA/RBDT.hxx:51]:
```suggestion
      if (!file || file->IsZombie()) {
```

File: tmva/tmva/inc/TMVA/RBDT.hxx [hunk 2]
```diff
@@ -48,6 +48,9 @@ public:
    {
       // Get number of output nodes of the forest
       auto file = TFile::Open(filename.c_str(), "READ");
```
Comment (eguiraud) [inline tmva/tmva/inc/TMVA/RBDT.hxx:50]: is the file leaked?
```suggestion
      std::unique_ptr<TFile> file{TFile::Open(filename.c_str())};
```

**Review by eguiraud — CHANGES_REQUESTED** [review]

File: tmva/tmva/inc/TMVA/RBDT.hxx (hunk 2 above)
Comment (lmoneta) [reply tmva/tmva/inc/TMVA/RBDT.hxx:50]: Yes, correct. It was only closed but not deleted the file ! Thanks for finding this!

**Review by eguiraud — APPROVED** [review]
LGTM. `GetObjectSafe` is type-unsafe, but that's for another PR :smile:

**Pull request merged**